### PR TITLE
Add fdroid build flavor

### PR DESCRIPTION
**Pull request by jspricke**
- Move GMS depending code to gms flavor.
- Add Wrappers for functions depending on GMS.
- Add empty implementations for fdroid flavor.
  TODO: provide non GMS implementations.

Closes: #70

### CODE REVIEW

**Comment by jspricke** [comment]
This is a first prototype to provide `fdroid` and `gms` flavors. I've tested the `fdroid` variant and it works (apart from localization). I don't have a phone with GMS to test the other variant, so it is purely compile tested.

File: app/src/fdroid/java/com/meteocool/location/ResolvableApiException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.meteocool.location;
+
+import android.app.Activity;
+
+public class ResolvableApiException extends Exception {
+    public void startResolutionForResult(Activity activity, int requestCode) {}
+}
```
Comment (Wheedman) [inline app/src/fdroid/java/com/meteocool/location/ResolvableApiException.java:7]: Bitte zu Kotlin konvertieren.

File: app/src/gms/java/com/meteocool/location/ResolvableApiException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.meteocool.location;
+
+import android.app.Activity;
+import com.google.android.gms.common.api.Status;
+
+public class ResolvableApiException extends com.google.android.gms.common.api.ResolvableApiException {
+	public ResolvableApiException(Status status) {
+		super(status);
+	}
+}
```
Comment (Wheedman) [inline app/src/gms/java/com/meteocool/location/ResolvableApiException.java:10]: Ebenfalls zu Kotlin konvertieren.

File: app/src/fdroid/java/com/meteocool/location/ResolvableApiException.java (hunk 1 above)
Comment (jspricke) [reply app/src/fdroid/java/com/meteocool/location/ResolvableApiException.java]: done.

File: app/src/gms/java/com/meteocool/location/ResolvableApiException.java (hunk 2 above)
Comment (jspricke) [reply app/src/gms/java/com/meteocool/location/ResolvableApiException.java]: done.

File: app/src/fdroid/java/com/meteocool/location/ResolvableApiException.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.meteocool.location
+
+import android.app.Activity
+
+class ResolvableApiException: Exception {
+    constructor()
```
Comment (Wheedman) [inline app/src/fdroid/java/com/meteocool/location/ResolvableApiException.kt:6]:
```suggestion
class ResolvableApiException: Exception() {
```

**Comment by Wheedman** [comment]
Besides that mark as ready for review and merge.

**Pull request merged**